### ISSUE TITLE
Support event payloads

### ISSUE THREAD

**Issue by chenlijun99** [StateSmith/StateSmith#372]
Oftentimes events can carry additional payload data (e.g. new UART message received). It would be cool for it to be supported by StateSmith.

**Comment by adamfk** [comment]
I agree. In the mean time, you can store the additional event information inside a state machine variable.

**Comment by chenlijun99** [comment]
Thanks for the workaround

**Comment by adamfk** [comment]
You are welcome :) I've used this work around a number of times successfully.

Even though it might take a while before we implement this feature, I'd be happy to hear ideas/thoughts on it.

**Comment by renestraub** [comment]
I second this request completely. I use this pattern all the time when working with state machines. Using a SM variable is not thread safe when more than one SM and/or IRQs are involved. In the end you have to create (thread/IRQ) safe queue to hold the payload.
